pkg/server/controller.go

```diff
@@ -129,6 +207,65 @@ func (ctrl *Controller) drainMiddleware(next http.HandlerFunc) http.HandlerFunc
 	}
 }
 
+func (ctrl *Controller) authMiddleware(next http.HandlerFunc) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		jwtCookie, err := r.Cookie(jwtCookieName)
+		if err != nil {
+			ctrl.httpServer.ErrorLog.Printf("There seems to be problem with jwt token cookie: %v", err)
+			http.Redirect(w, r, "/login", http.StatusTemporaryRedirect)
+			return
+		}
+
+		if jwtCookie == nil {
+			ctrl.httpServer.ErrorLog.Printf("Missing jwt cookie")
+			http.Redirect(w, r, "/login", http.StatusTemporaryRedirect)
+			return
+		}
+
+		jwtToken := jwtCookie.Value
+		token, err := jwt.Parse(jwtToken, func(token *jwt.Token) (interface{}, error) {
+			if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+				return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+			}
+			return []byte(ctrl.config.JWTSecret), nil
+		})
+
+		if err != nil {
+			ctrl.httpServer.ErrorLog.Printf("Error parsing jwt token: %v", err)
+			http.Redirect(w, r, "/login", http.StatusTemporaryRedirect)
+			return
+		}
+
+		claims, ok := token.Claims.(jwt.MapClaims)
+		if !ok || !token.Valid {
```

Comment:
`jwt.Parse` carries out all the necessary verifications. I guess this code won't be reached.